### PR TITLE
deriver for network state view, expose in snarkyjs

### PR DESCRIPTION
**Pull request by Trivo25**
allows SnarkyJS to pass in a custom network state and set custom network preconditions when evaluating transactions via `apply_json_transaction`

Instead of using a hard-coded network state, the state will now be maintained within SnarkyJS

accompanies https://github.com/o1-labs/snarkyjs/pull/329

### CODE REVIEW

**Review by mrmr1993 — APPROVED** [review]

File: src/lib/snarky_js_bindings/lib/snarky_js_bindings_lib.ml [hunk 1]
```diff
@@ -2870,6 +2870,16 @@ module Ledger = struct
...
+    let deriver =
+      Mina_base.Zkapp_precondition.Protocol_state.View.deriver
+      @@ Fields_derivers_zkapps.o ()
+    in
+    let of_json = Fields_derivers_zkapps.of_json deriver in
+    fun (json : Js.js_string Js.t) :
+        Mina_base.Zkapp_precondition.Protocol_state.View.t ->
+      json |> Js.to_string |> Yojson.Safe.from_string |> of_json
+
   let dummy_state_view : Mina_base.Zkapp_precondition.Protocol_state.View.t =
```
Comment (mitschabaude) [inline src/lib/snarky_js_bindings/lib/snarky_js_bindings_lib.ml:2883]: can we delete the `dummy_state_view` if it's unused?

**Review by mitschabaude — APPROVED** [review]

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mitschabaude** [comment]
ci-build-me!

**Comment by mitschabaude** [comment]
!ci-build-me

**Comment by mitschabaude** [comment]
I SAID BUILD ME!!!

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by Trivo25** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Pull request merged**